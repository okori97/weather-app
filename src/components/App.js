import "../styles/App.css";
import "../styles/index.css";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import SearchForm from "./SearchForm";
import React, { useState, useEffect } from "react";
import getForecasts from "../requests/getForecasts";

const App = () => {
  const [location, setLocation] = useState({});
  const [forecasts, setForecasts] = useState([]);
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  console.log(searchText);

  useEffect(() => {
    getForecasts(setLocation, setForecasts, setSelectedDate);
  }, []);

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const handleCitySelect = () => {
    getForecasts(setLocation, setForecasts, setSelectedDate, searchText);
  };
  return (
    <div className="weather-app">
      <div>
        <LocationDetails city={location.city} country={location.country} />
        <SearchForm
          onSubmit={handleCitySelect}
          searchText={searchText}
          setSearchText={setSearchText}
        />
        <ForecastSummaries
          forecasts={forecasts}
          onForecastSelect={handleForecastSelect}
        />
        {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
      </div>
    </div>
  );
};

export default App;
