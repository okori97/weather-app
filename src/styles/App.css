.country-name {
  font-size: 16px;
  color: #435180;
  font-weight: 400;
  margin: 0;
}

.location-name {
  font-size: 24px;
  font-weight: 700;
  margin-bottom: 0;
}

.weather-app {
  max-width: 100%;
  width: 850px;
  margin: 0 auto 0 auto;
}
