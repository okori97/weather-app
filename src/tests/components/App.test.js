import { render, screen } from "@testing-library/react";
import App from "../../components/App";
import forecast from "../../data/forecast.json";

describe("App", () => {
  test("renders react app properly", async () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
