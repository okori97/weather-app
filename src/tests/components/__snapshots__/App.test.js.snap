// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders react app properly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <div>
      <h1>
        undefined, undefined
      </h1>
      <form
        class="form"
      >
        <input
          class="form__input"
          id="form__input"
          placeholder="Search a city..."
          type="search"
        />
        <input
          class="form__button"
          type="button"
          value="Submit"
        />
      </form>
      <div
        class="forecastSummaries--list"
      />
    </div>
  </div>
</DocumentFragment>
`;
